packages/web/src/pages/upload-page/pages/FinishPage.tsx

```diff
@@ -157,33 +156,16 @@ export const FinishPage = (props: FinishPageProps) => {
     }
   }, [upload.tracks, uploadType])
 
-  const visitButtonPath = useMemo(() => {
-    switch (uploadType) {
-      case UploadType.INDIVIDUAL_TRACK:
-        return upload.tracks?.[0].metadata.permalink
-      case UploadType.ALBUM:
-      case UploadType.PLAYLIST:
-        return upload.completedEntity
-          ? collectionPage(
-              null,
-              null,
-              null,
-              upload.completedEntity.permalink,
-              uploadType === UploadType.ALBUM
-            )
-          : ''
-      default:
-        if (accountHandle && (!upload.tracks || upload.tracks.length > 1)) {
-          return profilePage(accountHandle)
-        } else {
-          return upload.tracks?.[0].metadata.permalink
-        }
-    }
-  }, [upload.completedEntity, upload.tracks, uploadType, accountHandle])
+  const visitButtonPath = useUploadCompletionRoute({
+    uploadType,
+    upload,
+    accountHandle
+  })
 
-  const dispatchVisitEvent = useCallback(() => {
+  const handleViewUpload = useCallback(() => {
     dispatch(make(Name.TRACK_UPLOAD_VIEW_TRACK_PAGE, { uploadType }))
-  }, [dispatch, uploadType])
+    onContinue()
```

Comment:
not sure i understand why this moved?

Reply:
existing bug i caught - we never clear state when you continue from finish page via this button